src/instruments/src/Common/simVars.tsx

```diff
@@ -0,0 +1,455 @@
+    }
+
+    /**
+     * This function will be called by the useSimVar hook through the context
+     * and notifies us that there is one listener less for this specific SimVar
+     * and unit combination.
+     */
+    const unregister: UnregisterSimVar = (name, unit, maxStaleness, global): void => {
+        const key = getKey(name, unit, global);
+        const old = listeners.current[key];
+        if (!Array.isArray(old) || old.length === 0) {
+            throw new Error('Attempted to unregisterHook with no known listener');
+        }
+        if (old.length === 1) {
+            // if we're unregistering the last entry, delete the array...
+            delete listeners.current[key];
+        } else {
+            // ...otherwise, filter out the first occurence of this value
+            const index = listeners.current[key].indexOf(maxStaleness || 0);
+            listeners.current[key] = listeners.current[key].splice(index, 1);
+        }
+    }
+
+    return (
+        <InternalProvider value={{
+            retrieve,
+            update,
+            register,
+            unregister,
+        }}
+        >
+            { children }
+        </InternalProvider>
+    );
+}
+
+/**
+ * The useSimVar hook provides an easy way to read and write SimVars from React.
+ *
+ * It's signature is similar to useState and it regularly refreshes the SimVar
+ * to ensure your React component stays in sync with the SimVar being modified
+ * from outside your component (like from other components, XML or SimConnect).
+ *
+ * You may optionally specify the maximum refresh interval. If the same SimVar
+ * is used in multiple places, this hook will automatically deduplicate those
+ * for maximum performance, rather than fetching the SimVar multiple times.
+ * Setting the SimVar will instantly cause it to be updated in all other places
+ * within the same React tree.
+ *
+ * @param name The name of the SimVar.
+ * @param unit The unit of the SimVar.
+ * @param maxStaleness The maximum time in milliseconds that may elapse before
+ * the next render will cause a SimVar refresh from the simulator. This
+ * parameter is only an upper bound! If another hook requests the same SimVar
+ * with a lower maxStaleness, this hook will also benefit from that and refresh
+ * the value more often.
+ *
+ * @example
+ * // the return value is the value itself and a setter, similar to useState
+ * const [v1, setV1] = useSimVar('L:AIRLINER_V1_SPEED', 'Knots');
+ *
+ * @example
+ * // only refresh the SimVar every 500ms (unless this SimVar is lower elsewhere)
+ * const [lightsTest] = useSimVar('L:XMLVAR_LTS_Test', 'Bool', 500);
+ *
+ * @returns {[*, (function(*): void)]}
+ *
+ * @see {@link useSplitSimVar} if your SimVar is set through a K event
+ * @see {@link useInteractionSimVar} if you emit an H event whenever your SimVar changes
+ * @see {@link useGlobalVar} if you have a Global Var instead
+ */
+export const useSimVar = (
+    name: string,
+    unit: UnitName,
```

Comment:
**suggestion**: Instead of the `normalizeUnitName` function. Consider limiting the valid unit names to a list of unit names, such that we'll get some consistency in how things they are written throughout our code.

Reply:
I do agree, however as part of this PR I don't want to enforce this, it's not even correctly typed yet (unit name just being a placeholder types as `any` for now). I think this will follow when we eventually clean up the typings.